src/main/java/org/testng/internal/TestResult.java

```diff
@@ -276,7 +277,7 @@ public void setParameters(Object[] parameters) {
     m_parameters = new Object[parameters.length];
     for (int i = 0; i < parameters.length; i++) {
       // Copy parameter if possible because user may change it later
-      if (parameters[i] instanceof Cloneable) {
+      if (canAttemptCloning(parameters[i]) ) {
         try {
           Method clone = parameters[i].getClass().getDeclaredMethod("clone");
           m_parameters[i] = clone.invoke(parameters[i]);

```

Comment:
Any idea why we don't use directly `m_parameters[i] = parameters[i].clone()` here?

Reply:
I dont know. But I believe we are resorting to `cloning()` only when an intent for that has been expressed by implementing the marker interface `Cloneable`. So in places wherein a parameter doesn't implement it, why invoke `clone()` and do a shallow cloning ?

Reply:
And to answer your question of why we aren't invoking the `clone()` method directly and using reflection to do it, well.. I dont have an answer for that. That is existing code. I didn't want to change a lot of things to fix a bug :) Trying to keep the impact to a minimum

Reply:
I found the answer: `clone()` is protected by default and it not visible.